crates/revm/src/evm_impl.rs

```diff
@@ -213,11 +212,13 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
             self.data
                 .subroutine
                 .balance_add(coinbase, coinbase_gas_price * (gas.spend() - gas_refunded));
+            gas.spend() - gas_refunded
         } else {
             // touch coinbase
             self.data.subroutine.load_account(coinbase, self.data.db);
             self.data.subroutine.balance_add(coinbase, U256::zero());
-        }
+            0
```

Comment:
Unsure if we still want to do the gas refund calculation stuff here? Does `crate::USE_GAS` toggle if we even record gas at all?

Reply:
Good q. @rakita wdyt?

Reply:
This is fine, USE_GAS in general is to disable gas calculation if evm is needed in some wasm env where pricing is done differently.